scala/lambdas/ingest-event-aggregator/src/main/scala/uk/gov/nationalarchives/eventaggregator/Aggregator.scala

```diff
@@ -0,0 +1,104 @@
+package uk.gov.nationalarchives.eventaggregator
+
+import cats.Parallel
+import cats.effect.syntax.all.*
+import cats.effect.{Async, Outcome, Ref}
+import cats.syntax.all.*
+import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage
+import io.circe.generic.auto.*
+import io.circe.parser.decode
+import io.circe.syntax.*
+import io.circe.{Decoder, Encoder, Printer}
+import org.typelevel.log4cats.SelfAwareStructuredLogger
+import org.typelevel.log4cats.slf4j.Slf4jFactory
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue
+import uk.gov.nationalarchives.DADynamoDBClient.DADynamoDbWriteItemRequest
+import uk.gov.nationalarchives.eventaggregator.Aggregator.NewBatchReason.*
+import uk.gov.nationalarchives.eventaggregator.Aggregator.{Input, SFNArguments}
+import uk.gov.nationalarchives.eventaggregator.Lambda.{Batch, Config}
+import uk.gov.nationalarchives.{DADynamoDBClient, DASFNClient}
+
+import java.net.URI
+import java.time.Instant
+import java.util.UUID
+
+trait Aggregator[F[_]]:
+
+  def aggregate(config: Config, ref: Ref[F, Map[String, Batch]], messages: List[SQSMessage], remainingTimeInMillis: Int)(using
+      Encoder[SFNArguments],
+      Decoder[Input]
+  ): F[List[Outcome[F, Throwable, Unit]]]
+
+object Aggregator:
+
+  case class Input(id: UUID, location: URI)
+
+  case class SFNArguments(groupId: String, batchId: String, waitFor: Long, retryCount: Int = 0)
```

Comment:
I think we should try and get the types to signify what is there rather than making longer variable names. I've been playing around with opaque types and I got something like this.
```scala
object Duration:
    opaque type Seconds = Int

    object Seconds:
      def apply(length: Int): Seconds = length

    extension (s: Seconds)
      def length: Int = s
  end Duration
    
  extension (i: Int)
    private def seconds: Seconds = Seconds(i) // Can just do private def seconds: Seconds = i because it won't compile
 ```
This lets us pass in `1.seconds` using this extension method which returns a Seconds object but we can't do:
```scala
import scala.concurrent.duration.*
SFNArguments("", "", 1.days, 0)
```
This way, it's clear the that waitFor is for seconds and we have to explicitly create a Seconds object when creating the case class.